### ISSUE TITLE
API: Running a job in 2.6.0 throws error when header 'Accept: application/json' is included but body is not json

### ISSUE THREAD

**Issue by rmbolger** [rundeck/rundeck#1504]
I'm using the _/api/12/job/[ID]/executions_ endpoint to run a job with curl.  If I send the body with header 'Content-Type: application/x-www-form-urlencoded' and also include the 'Accept: application/json' header to request a json response, the API throws an 'api.error.item.unsupported-format' error with message "The format is not valid: json".

The documentation for API v14 indicates  endpoint expects a json body when the header 'Content-Type: application/json' is passed.  I interpreted that to mean I optionally still pass the body as 'Content-Type: application/x-www-form-urlencoded' and still request a json response with 'Accept: application/json'.  But that doesn't seem to be the case.

If I exclude the 'Accept: application/json' header, the request works. But the response is not json output either.

**Comment by gschueler** [comment]
Json is not supported in API v12.  Use `/api/14/` in the URL

**Comment by rmbolger** [comment]
Explicitly specifying v14 in the URL does fix the problem when I'm already running v14.  My problem is I'm in the midst of a migration from v12 to v14.  So I'm trying to make sure existing code that works against v12 continues to work against v14 without modification until all of our instances are v14.

In v12, it would simply ignore the 'Accept: application/json' header instead of throwing an error. I was hoping it would also ignore it in v14 when I specified v12 in the URL.  For now, I've worked around my problem by updating the existing code to not specify the accept json header which works as expected on both v12 and v14 when specifying 12 in the URL. In retrospect, I should have done that from the beginning.
